tests/functionality_tests/sensor_physical_testing.py

```diff
@@ -0,0 +1,42 @@
+from engine.robot_state import Robot_State
+from engine.robot import Robot
+from electrical.gps import GPS
+from engine.grid import Grid
+from engine. database import DataBase
+import math
+
+if __name__ == "__main__":
+    # make sure to go to imu and gps and set import to true
+    test_state = 0  # 0 is no node straight, 1 is node straight, 2 is turn, -1 is sensor
+    rb_state = Robot_State(0, 0, 0, .2, .5, .2)
+    rb = Robot(rb_state)
+    database = DataBase(rb)
+    rb.execute_setup()
+    rb_state.track_obstacle_thread.stop()
+
+    if test_state == -1:
+        print("gps")
```

Comment:
you can make these print statements nicer, a cool trick: sending multiple arguments into print (and u dont have to cast anything to a string!)
print("gps", rb_state.gps.get_gps())

Reply:
will change